### PR TITLE
add renametokengroup function

### PR DESCRIPTION
**Pull request by MightyDevKnight**

### CODE REVIEW

File: src/app/components/TokenGroupHeading.tsx [hunk 1]
```diff
@@ -3,34 +3,97 @@ import { useSelector } from 'react-redux';
...
+        <ContextMenuContent>
+          <ContextMenuItem disabled={editProhibited} onSelect={handleDelete}>
+            Delete
+          </ContextMenuItem>
+          <ContextMenuItem disabled={editProhibited} onSelect={handleRename}>
+            Rename
+          </ContextMenuItem>
+        </ContextMenuContent>
+      </ContextMenu>
+      <Modal isOpen={showNewGroupNameField} close={() => setShowNewGroupNameField(false)}>
```
Comment (LiamMartens) [inline src/app/components/TokenGroupHeading.tsx:66]: Could these handlers be moved to `useCallback`

File: src/app/components/TokenGroupHeading.tsx [hunk 2]
```diff
@@ -3,34 +3,97 @@ import { useSelector } from 'react-redux';
...
+        <Stack direction="column" justify="center" gap={4} css={{ textAlign: 'center' }}>
+          <Heading size="small">Rename {path.split('.').pop()}</Heading>
```
Comment (LiamMartens) [inline src/app/components/TokenGroupHeading.tsx:68]: Should probably memo this heading label, the `split` operation will create a new array each render cycle

File: src/app/components/TokenGroupHeading.tsx [hunk 3]
```diff
@@ -3,34 +3,97 @@ import { useSelector } from 'react-redux';
...
+          <Heading size="small">Renaming only affects tokens of the same type</Heading>
+          <form onSubmit={handleRenameTokenGroupSubmit}>
+            <Stack direction="column" gap={4}>
+              <Input
+                full
+                onChange={(e) => setNewTokenGroupName(e.target.value)}
```
Comment (LiamMartens) [inline src/app/components/TokenGroupHeading.tsx:74]: same about the callback

File: src/app/components/TokenGroupHeading.tsx [hunk 4]
```diff
@@ -3,34 +3,97 @@ import { useSelector } from 'react-redux';
...
+                type="text"
+                name="tokengroupname"
+                required
+                defaultValue={path.split('.').pop() || ''}
```
Comment (LiamMartens) [inline src/app/components/TokenGroupHeading.tsx:78]: Same about memoizing it - especially since we are using it multiple times

**Review by LiamMartens — CHANGES_REQUESTED** [review]

File: src/app/components/TokenGroupHeading.tsx (hunk 1 above)
Comment (MightyDevKnight) [reply src/app/components/TokenGroupHeading.tsx]: Done.

Comment (MightyDevKnight) [reply src/app/components/TokenGroupHeading.tsx]: Done.

Comment (MightyDevKnight) [reply src/app/components/TokenGroupHeading.tsx]: Done.

Comment (MightyDevKnight) [reply src/app/components/TokenGroupHeading.tsx]: Done.

Comment (MightyDevKnight) [reply src/app/components/TokenGroupHeading.tsx]: Done.

File: src/app/components/TokenGroupHeading.tsx (hunk 4 above)
Comment (MightyDevKnight) [reply src/app/components/TokenGroupHeading.tsx]: Done.

File: src/app/components/TokenGroupHeading.tsx (hunk 3 above)
Comment (MightyDevKnight) [reply src/app/components/TokenGroupHeading.tsx]: Done.

File: src/app/components/TokenGroupHeading.tsx (hunk 2 above)
Comment (MightyDevKnight) [reply src/app/components/TokenGroupHeading.tsx]: Done.

File: src/app/components/TokenGroupHeading.tsx [hunk 5]
```diff
@@ -63,29 +75,29 @@ export default function TokenGroupHeading({ label, path, id, type }: Props) {
           </ContextMenuItem>
         </ContextMenuContent>
       </ContextMenu>
-      <Modal isOpen={showNewGroupNameField} close={() => setShowNewGroupNameField(false)}>
+      <Modal isOpen={showNewGroupNameField} close={() => handleSetNewTokenGroupNameFileClose}>
```
Comment (LiamMartens) [inline src/app/components/TokenGroupHeading.tsx:78]: This should just be `<Modal isOpen={showNewGroupNameField} close={handleSetNewTokenGroupNameFileClose}>`

File: src/app/components/TokenGroupHeading.tsx [hunk 6]
```diff
@@ -63,29 +75,29 @@ export default function TokenGroupHeading({ label, path, id, type }: Props) {
...
         <Stack direction="column" justify="center" gap={4} css={{ textAlign: 'center' }}>
-          <Heading size="small">Rename {path.split('.').pop()}</Heading>
+          <Heading size="small">Rename {oldTokenGoupName}</Heading>
           <Heading size="small">Renaming only affects tokens of the same type</Heading>
           <form onSubmit={handleRenameTokenGroupSubmit}>
             <Stack direction="column" gap={4}>
               <Input
                 full
-                onChange={(e) => setNewTokenGroupName(e.target.value)}
+                onChange={(e) => handleNewTokenGroupNameChange(e)}
```
Comment (LiamMartens) [inline src/app/components/TokenGroupHeading.tsx:86]: This should be `onChange={handleNewTokenGroupNameChange}`

**Review by LiamMartens — CHANGES_REQUESTED** [review]

File: src/app/components/TokenGroupHeading.tsx (hunk 5 above)
Comment (MightyDevKnight) [reply src/app/components/TokenGroupHeading.tsx]: Done

File: src/app/components/TokenGroupHeading.tsx (hunk 6 above)
Comment (MightyDevKnight) [reply src/app/components/TokenGroupHeading.tsx:87]: Done

**Review by LiamMartens — APPROVED** [review]

**Pull request closed without merging**